src/auth/auth.service.ts

```diff
@@ -107,52 +112,56 @@ export class AuthService {
 
     const text = `Welcome to the application. To confirm the email address, click here: ${url}`;
 
-    await this.emailService.sendMail({
+    const sendMailRequestdata: SendMailRequestData = {
       to: email,
       subject: 'Email confirmation',
       text,
-    });
+    };
+
+    await this.emailService.sendMail(sendMailRequestdata);
+
     return {
       message: EmailMessages.emailSendSuccess,
     };
   }
-  async verify(token: string) {
+  async verify(verifyTokenRequestData: VerifyTokenRequestData) {
+    const token = verifyTokenRequestData.token;
+
     const payload = this.jwtService.verify<EmailVerificationPayload>(token, {
       secret: process.env.JWT_VERIFICATION_TOKEN_SECRET,
     });
 
-    const record = await this.emailVerificationRepo.findOneByOrFail({
-      token,
+    const record = await this.emailVerificationRepo.findOne({
+      where: {
+        token,
+        expiresAt: MoreThan(new Date()),
+      },
     });
 
-    if (record.expiresAt < new Date()) {
+    if (!record) {
       throw new BadRequestException('Token has expired');
     }
 
     await this.emailVerificationRepo.save(record);
 
-    const user = await this.userRepository.findOne({
-      where: {
-        email: payload.email,
-      },
-    });
+    const user = await this.userService.findOneByField('email', payload.email);
     if (!user) throw new Error('User not found');
```

Comment:
Maybe `NotFoundException` could also be suitable here?

Also this structure could provide more readability :

```suggestion
    if (!user){
     throw new Error('User not found');
    }
```

Reply:
updated